app/managers/performance_metrics_manager/performance_metrics_batcher.ts

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {AppState, Platform, type AppStateStatus} from 'react-native';
+
+import {sendPerformanceReport} from '@actions/remote/performance';
+import DatabaseManager from '@database/manager';
+import {getConfigValue} from '@queries/servers/system';
+import {toMilliseconds} from '@utils/datetime';
+import {logDebug} from '@utils/log';
+
+const MAX_BATCH_SIZE = 100;
+const INTERVAL_TIME = toMilliseconds({seconds: 60});
+
+class Batcher {
+    private started = false;
+    private batch: PerformanceReportMeasure[] = [];
+    private serverUrl: string;
+    private sendTimeout: NodeJS.Timeout | undefined;
+    private lastAppStateIsActive = AppState.currentState === 'active';
+
+    constructor(serverUrl: string) {
+        this.serverUrl = serverUrl;
+        AppState.addEventListener('change', (appState) => this.onAppStateChange(appState));
+    }
+
+    private start() {
+        this.started = true;
+        clearTimeout(this.sendTimeout);
+        this.sendTimeout = setTimeout(() => this.sendBatch(), INTERVAL_TIME);
```

Comment:
Should this use a `setInterval`? I didn't use a `setInterval` on web because of the random jitter I added to the interval time, but I don't think we need that on mobile because sessions aren't as long-lived

Reply:
Oh, I see. We're doing this with a one-off interval because we aren't leaving the interval running in the background